src/NATS.Client.Core/NatsSub.cs

```diff
@@ -9,6 +11,7 @@ internal NatsSubBase(NatsConnection connection, SubscriptionManager manager, str
     {
         Connection = connection;
         Manager = manager;
+        HeaderParser = new HeaderParser(Encoding.UTF8);
```

Comment:
Header Parser is thread safe, right?  Should we can allocate 1 per connection as opposed to 1 per subscription.

Also, the default encoding is technically ASCII since this follows the HTTP RFC

Certain clients enforce ASCOO while others do not though.  For example the Go implementation defaults to UTF-8.

So I think that we should add `HeaderEncoding` to `NatsOptions`, default it to `Encoding.ASCII` there but let someone change it if they want.

Reply:
All makes sense.